File: model/src/main/java/io/github/chriszhong/model/organization/entity/PolicyFactory.java
package io.github.chriszhong.model.organization.entity;

import io.github.runtimemodels.chazm.entity.Agent;
import io.github.runtimemodels.chazm.entity.Policy;
import io.github.runtimemodels.chazm.id.UniqueId;

/**
 * The {@linkplain PolicyFactory} interface defines the APIs for constructing {@linkplain Policy}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PolicyFactory {

	/**
	 * Constructs a {@linkplain Policy}.
	 *
	 * @param id
	 *            the {@linkplain UniqueId} that represents the {@linkplain Policy}.
	 * @return a {@linkplain Agent}.
	 */
	Policy buildPolicy(UniqueId<Policy> id);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/factory/entity/PolicyFactory.kt
package runtimemodels.chazm.model.factory.entity

import runtimemodels.chazm.api.entity.Policy
import runtimemodels.chazm.api.id.PolicyId
import runtimemodels.chazm.api.id.UniqueId

/**
 * The [PolicyFactory] interface defines the APIs for constructing [Policy]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PolicyFactory {

    /**
     * Constructs a [Policy].
     *
     * @param id the [UniqueId] that represents the [Policy].
     * @return a [Policy].
     */
    fun build(id: PolicyId): Policy

}
